### PR TITLE
git: Fix test

### PR DESCRIPTION
**Pull request by DoomHammer**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)? 

-----
As Travis environment does not use `git` for `HOMEBREW_REPOSITORY` a more generic set of tests is needed.

`brew audit` is not passing same way as before this change.

### CODE REVIEW

File: Formula/git.rb [hunk 1]
```diff
@@ -169,8 +170,14 @@ def caveats; <<-EOS.undent
   end
 
   test do
-    HOMEBREW_REPOSITORY.cd do
-      assert_equal "bin/brew", `#{bin}/git ls-files -- bin`.strip
-    end
+    assert_match(/^git version .*$/, `#{bin}/git --version`.strip)
+    assert `#{bin}/git init`
+    assert `#{bin}/git status`
+    assert `#{bin}/git config --global user.email "you@example.com"`
```
Comment (sjackman) [inline Formula/git.rb]: `--global` modifies `~/.gitconfig`. Although the `test` resets `$HOME`, it still makes me nervous. Please remove `--global`.

File: Formula/git.rb [hunk 2]
```diff
@@ -169,8 +170,14 @@ def caveats; <<-EOS.undent
   end
 
   test do
-    HOMEBREW_REPOSITORY.cd do
-      assert_equal "bin/brew", `#{bin}/git ls-files -- bin`.strip
-    end
+    assert_match(/^git version .*$/, `#{bin}/git --version`.strip)
+    assert `#{bin}/git init`
```
Comment (sjackman) [inline Formula/git.rb]: Use `system "foo"` rather than
```
assert `foo`
```

File: Formula/git.rb [hunk 3]
```diff
@@ -169,8 +170,14 @@ def caveats; <<-EOS.undent
   end
 
   test do
-    HOMEBREW_REPOSITORY.cd do
-      assert_equal "bin/brew", `#{bin}/git ls-files -- bin`.strip
-    end
+    assert_match(/^git version .*$/, `#{bin}/git --version`.strip)
```
Comment (sjackman) [inline Formula/git.rb]: Use `shell_output("foo")` rather than
```
`foo`
```

**Comment by sjackman** [comment]
Much improved, Piotr. Thanks. After making the requested fixes, merge it. You may also want to send this fix upstream to Homebrew.

**Comment by sjackman** [comment]
Please use the PR subject format `git: short message here` so that `build-bottle-pr` (and us) can easily find the PR.

**Comment by DoomHammer** [comment]
So, should this be merged here first, then sent upstream or just sent to upstream and then we wait for merge?

**Comment by sjackman** [comment]
Looks great. Merge it here and send the fix upstream. Please use `git cherry-pick` to keep the commit author, date, log message etc identical to the Linuxbrew commit.

**Comment by DoomHammer** [comment]
`git cherry-pick` also carries the information that the commit closes Linuxbrew issue. Is it necessary for the upstream?

**Comment by sjackman** [comment]
Not really. Delete that.

**Comment by mikemcquaid** [comment]
> As Travis environment does not use git for HOMEBREW_REPOSITORY a more generic set of tests is needed.

Why does it not?

**Comment by sjackman** [comment]
It uses `curl` rather than `git` to download `brew` when testing `homebrew-core` to test that it's possible to install Linuxbrew without requiring `git`, which is a goal of Linuxbrew. `homebrew-core` is nonetheless cloned using `git`, so it's not a perfect test.

**Comment by mikemcquaid** [comment]
Ok, thanks.

**Pull request closed without merging**